frb_rust/src/rust2dart.rs

```diff
@@ -77,37 +75,132 @@ impl TaskCallback {
     pub fn stream_sink<T: IntoDart>(&self) -> StreamSink<T> {
-        StreamSink::new(self.rust2dart)
+        StreamSink::new(self.rust2dart.clone())
     }
 }
 
+#[derive(Clone)]
+pub struct Channel(pub String);
+
+impl Channel {
+    #[cfg(wasm)]
+    pub fn port(&self) -> MessagePort {
+        PortLike::broadcast(&self.0)
+    }
+}
+
+// #[cfg(wasm)]
+// #[derive(Default)]
+// struct Signal {
+//     started: std::sync::Mutex<bool>,
+//     cvar: std::sync::Condvar,
+// }
+
 /// A sink to send asynchronous data back to Dart.
 /// Represented as a Dart
 /// [`Stream`](https://api.dart.dev/stable/dart-async/Stream-class.html).
 #[derive(Clone)]
 pub struct StreamSink<T: IntoDart> {
+    #[cfg(not(wasm))]
     rust2dart: Rust2Dart,
+    /// On WASM, [`Rust2Dart`] is not [`Send`] so we only
+    /// carry the name of the broadcast channel.
+    #[cfg(wasm)]
+    channel: Channel,
+    // #[cfg(wasm)]
+    // signal: std::sync::Arc<Signal>,
     _phantom_data: PhantomData<T>,
 }
 
+// #[cfg(wasm)]
+// lazy_static::lazy_static! {
+//     static ref SINK_INIT: std::sync::Mutex<std::collections::HashMap<String, std::sync::Arc<Signal>>> = Default::default();
+// }
+
+// #[cfg(wasm)]
+// #[doc(hidden)]
+// #[wasm_bindgen]
+// pub fn __start_streamsink(name: String) {
+//     use crate::pool::WorkerPool;
+//     thread_local! {
+//         static CONTROL_POOL: WorkerPool = WorkerPool::new(1, script_path().unwrap()).unwrap();
+//     }
+
+//     CONTROL_POOL.with(|pool| {
+//         pool.run(transfer!(|| {
+//             if let Some(signal) = SINK_INIT.lock().unwrap().get(&name) {
+//                 *signal.started.lock().unwrap() = true;
+//                 signal.cvar.notify_all();
+//             } else {
+//                 console_error(&format!("StreamSink was not registered in time: {}", name));
+//             }
+//         }))
+//         .unwrap();
+//     });
+// }
+
 impl<T: IntoDart> StreamSink<T> {
     /// Create a new sink from a port wrapper.
     pub fn new(rust2dart: Rust2Dart) -> Self {
+        #[cfg(wasm)]
+        let name = rust2dart
+            .isolate
+            .broadcast_name()
+            .expect("Not a BroadcastChannel");
         Self {
+            #[cfg(not(wasm))]
             rust2dart,
+            #[cfg(wasm)]
+            channel: Channel(name),
+            // #[cfg(wasm)]
+            // signal: {
+            //     let mut signals = SINK_INIT.lock().unwrap();
+            //     if signals.contains_key(&name) {
+            //         drop(signals);
+            //         panic!("Name collision: {}", name);
+            //     }
+            //     signals.entry(name).or_default().clone()
+            // },
             _phantom_data: PhantomData,
         }
     }
 
+    fn rust2dart(&self) -> Rust2Dart {
+        #[cfg(not(wasm))]
+        return self.rust2dart.clone();
+
+        #[cfg(wasm)]
+        Rust2Dart::new(self.channel.port())
+    }
```

Comment:
nit: maybe we can refactor the code a bit: Currently it is 

```
pub struct StreamSink<T: IntoDart> {
    #[cfg(not(wasm))]
    rust2dart: Rust2Dart,
    #[cfg(wasm)]
    channel: Channel,
    ... other fields ...
}
```

But maybe we indeed want:

```
#[not(wasm)]
struct SendableRust2Dart(Rust2Dart);
#[wasm]
struct SendableRust2Dart(Channel);

pub struct StreamSink<T: IntoDart> {
    rust2dart: SendableRust2Dart,
    ... other fields ...
}
```